### ISSUE TITLE
incorrect limit pushdown with broadcast joins

### ISSUE THREAD

**Issue by mtuncer** [citusdata/citus#2027]
Limit pushdown logic is incorrectly applied to queries that contain a broadcast join.

Consider following query
```sql
SELECT
	l_orderkey,
	avg(o_totalprice / l_quantity) AS unit_price
FROM
	lineitem,
	orders
WHERE
	l_orderkey = o_custkey
GROUP BY
	l_orderkey
ORDER BY 2 DESC, 1 DESC
LIMIT 5;
```

Both tables have 2 shards each. They are range distributed tables on ```l_orderkey``` and ```o_orderkey``` columns respectively.

When are checking if we should pushdown limit (and order by), we check if the query is on a disjoint partition key (i.e. group by is on partition column).  We find that group by is on l_orderkey column, and it is partition key for the distributed table, and push down the limit and order by to worker query.

This logic is not correct, even though there is a group by clause on partition column. Reason is that we are doing broadcast join,  we are joining the shards of the first table (lineitem) with each shard from the second (orders) table.

we have 4 worker queries joining  lineitem_1 x orders_1, lineitem_1 x orders_2, lineitem_2 x orders_1, and lineitem_2 x orders_2.

Pushing down limits in such query plan may cause returning  wrong results.

**Comment by metdos** [comment]
Thanks @mtuncer, we will ditch broadcast joins with the next major release. Still, we wanted to document this.

**Comment by mtuncer** [comment]
@metdos  we close this now

**Comment by velioglu** [comment]
Fixed by #1980
